app/assets/javascripts/controllers/ansible_credentials/ansible_credentials_form_controller.js

```diff
@@ -0,0 +1,130 @@
+      API.get('/api/providers?collection_class=ManageIQ::Providers::EmbeddedAutomationManager')
+         .then(setManagerResource)
+         .catch(miqService.handleFailure);
+
+      vm.afterGet = true;
+      vm.modelCopy = angular.copy( vm.credentialModel );
+      miqService.sparkleOff();
+    }
+  };
+
+  vm.cancelClicked = function(angularForm) {
+    if (credentialId == 'new') {
+      getBack(__("Creation of new Credential was canceled by the user."), true);
+    } else {
+      getBack(sprintf(__("Edit of Credential \"%s\" was canceled by the user."), vm.credentialModel.name), true);
+    }
+  };
+
+  vm.resetClicked = function(angularForm) {
+    vm.credentialModel = angular.copy( vm.modelCopy );
+    angularForm.$setPristine(true);
+    miqService.miqFlash("warn", __("All changes have been reset"));
+  };
+
+  vm.saveClicked = function(angularForm) {
+    API.put('/api/authentications/' + credentialId, vm.credentialModel)
+       .then(getBack(sprintf(__("Modification of Credential \"%s\" has been successfully queued."), vm.credentialModel.name), false))
+       .catch(miqService.handleFailure);
+  };
+
+  vm.addClicked = function(angularForm) {
+    addCredentialKind();
+    API.post('/api/authentications/', vm.credentialModel)
+       .then(getBack(sprintf(__("Add of Credential \"%s\" has been successfully queued."), vm.credentialModel.name)))
+       .catch(miqService.handleFailure);
+  };
+
+  function getCredentialOptions(response) {
+    Object.assign(vm.credential_options, response.data.credential_types.embedded_ansible_credential_types);
+
+    for (opt in vm.credential_options) {
+      vm.select_options.push({'value': opt, 'label': vm.credential_options[opt].label});
+    }
+  }
+
+  function getCredentialFormData(response) {
+    vm.credentialModel.id = response.id;
+    vm.credentialModel.name = response.name;
+    vm.credentialModel.type = response.type;
+
+    // we need to merge options and vm.credentialModel
+    for (var opt in response.options) {
+      var item = vm.credential_options[vm.credentialModel.type]['attributes'][opt]
+
+      // void the password fields first
+      if (item.hasOwnProperty('type') && item['type'] == 'password') {
+        vm.credentialModel[opt] = '';
+      }
+      else {
+        vm.credentialModel[opt] = response.options[opt];
+      }
+    }
+
+    vm.modelCopy = angular.copy( vm.credentialModel );
+    vm.afterGet = true;
+    miqService.sparkleOff();
+  }
+
+  function getBack(message, warning = false, error = false) {
+    var url = '/ansible_credential/show_list' + '?flash_msg=' + message + '&escape=true';
+
+    if (warning) {
+      url += '&flash_warning=true&flash_error=false';
+    }
+    else if (error) {
+      url += '&flash_warning=false&flash_error=true';
+    }
+
+    $window.location.href = url;
+  }
+
+  function setManagerResource(response) {
+    vm.credentialModel.manager_resource = { "href": response.resources[0].href };
```

Comment:
The point of *embedded* ansible is that it's embedded in the appliance, so there's only one, ever.

The API returns a list because it's the API and we want it consistent with all the other API calls, but AFAIK there can't ever be more than one embedded ansible.